apis/r/R/TileDBArray.R

```diff
@@ -92,19 +93,19 @@ TileDBArray <- R6::R6Class(
 
     #' @description Add list of metadata to the specified TileDB array. (lifecycle: maturing)
     #' @param metadata Named list of metadata to add.
+    #' @param timestamps Optional two-element vector of timestamp range starts and end
     #' @return NULL
-    set_metadata = function(metadata) {
+    set_metadata = function(metadata, timestamps = NULL) {
       stopifnot(
         "Metadata must be a named list" = is_named_list(metadata)
       )
 
-      private$check_open_for_write()
+      #private$check_open_for_write()
 
       for (nm in names(metadata)) {
-          #spdl::warn("[set_metadata] key {}", nm)
           val <- metadata[[nm]]
-          spdl::debug("[set_metadata] setting key {} to {} ({})", nm, val, class(val))
-          set_metadata(self$uri, nm, val, class(val), TRUE, soma_context())
+          spdl::debug("[TileDBArray$set_metadata] setting key {} to {} ({})", nm, val, class(val))
+          set_metadata(self$uri, nm, val, class(val), TRUE, soma_context(), timestamps)
```

Comment:
We should be using the cached `self$tiledb_timestamp` here (range form: `c(self$tiledb_timestamp, self$tiledb_timestamp)`)

Reply:
Addressed in #2930